### PR TITLE
Fixes Travis and introduces usage of PEP 484

### PR DESCRIPTION
**Pull request by djperrefort**
~This PR introduces the usage of PEP 484 to most of the package, and also fixes the Travis/RTD config files so the docs and tests are running again.~

This PR introduces the usage of PEP 484 to most of the package, and also fixes the Travis config files so the tests are running automatically again. I thought I had fixed the ReadTheDocs build, but we are still using too many installation resources and RTD is killing the build. It seems like installing `tensorflow` is the problem. I don't have a clean solutuion for this so I'll think on it more, but the remaining edits still warrant their own PR.

### CODE REVIEW

File: broker/ztf_archive/_download_data.py [hunk 1]
```diff
@@ -97,17 +101,17 @@ def _download_alerts_file(file_name, out_dir, block_size, verbose):
             data.extractall(out_dir)
 
 
-def download_data_date(year, month, day, block_size=1024, verbose=True):
+def download_data_date(year, month, day, block_size=1024, verbose=True) -> None:
```
Comment (troyraen) [inline broker/ztf_archive/_download_data.py]: Is there a reason these args don't have a type specified? This syntax is new to me, but based on the other functions I think this should be:

```
def download_data_date(
        year: int, 
        month: int, 
        day: int, 
        block_size: int = 1024, 
        verbose: bool = True) -> None:
```

File: docs/source/index.rst [hunk 2]
```diff
@@ -4,7 +4,7 @@ What is Pitt Broker
 ===================
 
 **Pitt Broker** is a cloud-based, alert distribution service designed to provide
```
Comment (troyraen) [inline docs/source/index.rst]: I suggest we change "Pitt Broker" to "Pitt-Google Broker" (4 instances in this file).

Comment (djperrefort) [reply docs/source/index.rst]: Agreed

File: broker/ztf_archive/_download_data.py (hunk 1 above)
Comment (djperrefort) [reply broker/ztf_archive/_download_data.py]: Correct, I must have just missed those.

**Pull request merged**